### DISCUSSION TITLE
Bulldoze while track building

### DISCUSSION THREAD

**Discussion by Nihildum** [voxeltycoon/issues#332] [General]
Currently bulldozing while build a track is quite cumbersome. Each time I want to remove something, I have to select bulldoze and then I have to go back into the building menue. This are 5 clicks and each time you need to move the mouse two times away from the position where you currently are building. 

It would help if you could undo your last build actions (also with a full refound :)
or if the bulldoze action could be added to the work/menue stack.

One option would be:
 - Build a track
 - Hold shift (or an other modifier key) to enter bulldoze mode
 - Remove the tracks
 - Release shift
 - Continue building tracks

Another option is:
 - Build a track
 - Hit the bulldoze shortcut (currently /)
 - Remove the tracks
 - Hit the right mouse button or escape to drop the bulldoze mode
 - Continue building tracks

With both approaches the user would no longer need to move the mouse away while building tracks.

**Comment by JezzyBelle** [comment]
I agree... A last placement undo would be nice. 

I also would like a little better option for speed when placing rails.
I have found that...  
    If you place rail only a few spaces at a time when going through a grade it will plow or fill the grade at the same level you are currently placing track. It's a little slow using this method but then, it is costly so maybe it's a good thing. Even though it is only about 4 blocks max length that this works (seems 2 sometimes in very deep cuts),  it is a pretty simple method, making it easier to place a level grade as you go along.

**Comment by IDragonfire** [comment]
I have the same problem and this is maybe one solution.
It is anoying if you have to select a different category of the toolbar.

Similar problem, if you build a new factory complex and you have to switch between conveyors and the factory category in the toolbar.

PS.: Maybe a Quick Insert/Add is an alterantive option, like here:
https://blogs.mathworks.com/simulink/2019/01/16/predictive-quick-insert/

**Comment by gralewski** [comment]
Please don't invent the wheel again.

**Control** to enter **temporarily** buldoze mode. We have this perfect solution in OpenTTD already.
**Ctrl+Z** to undo placement.
